src/Examine/LuceneEngine/Providers/LuceneIndex.cs

```diff
@@ -1185,16 +1240,19 @@ private IndexWriter CreateIndexWriter()
             // ensure that it's unlocked here in that case.
             try
             {
-                if (IndexWriter.IsLocked(dir))
+                if (IndexWriter.IsLocked(dir) && (!(dir is ExamineDirectory) ||
+                                                  dir is ExamineDirectory examineDirectory &&
+                                                  !examineDirectory.IsReadOnly))
                 {
                     //unlock it!
                     IndexWriter.Unlock(dir);
                 }
             }
             catch (Exception ex)
             {
-                OnIndexingError(new IndexingErrorEventArgs(this, "The index was locked and could not be unlocked", null, ex));
-                return null;
+                OnIndexingError(new IndexingErrorEventArgs(this, "The index was locked and could not be unlocked", null,
+                    ex));
+                if (!isReadonly) return null;
```

Comment:
Can you add comments about these checks here for isReadOnly? Why do we ignore unlocking when that is the case, why do we not exit here when that is the case?

Reply:
@Shazwazza that is actually left from my testing and should be removed, good catch! :)